src/lib/components/ConfirmTransactions.js

```diff
@@ -12,78 +17,201 @@ ${json}
 \`\`\``;
 };
 
+const StorageDomain = {
+  page: "confirm_transactions",
+};
+
+const StorageType = {
+  SendTransactionWithoutConfirmation: "send_transaction_without_confirmation",
+};
+
 export default function ConfirmTransactions(props) {
   const gkey = useState(uuid());
   const near = useNear(props.networkId);
+  const accountId = useAccountId(props.networkId);
+  const cache = useCache();
+
   const [loading, setLoading] = useState(false);
 
+  const [transactions] = useState(props.transactions);
+  const [dontAskForConfirmation, setDontAskForConfirmation] = useState(null);
+  const [dontAskAgainChecked, setDontAskAgainChecked] = useState(false);
+  const [dontAskAgainErrorMessage, setDontAskAgainErrorMessage] = useState(null);
+
+  const widgetSrc = props.widgetSrc;
+
+  const getWidgetContractPermission = async (widgetSrc, contractId) =>
+    await cache.asyncLocalStorageGet(StorageDomain, {
+      widgetSrc,
+      contractId,
+      type: StorageType.SendTransactionWithoutConfirmation,
+    });
+
+  useEffect(() => {
+    (async () => {
+      if (transactions.length === 1) {
+        const contractId = transactions[0].contractName;
+        const isSignedIntoContract = await near.isSignedIntoContract(contractId);
+
+        const widgetContractPermission = await getWidgetContractPermission(widgetSrc, contractId);
+
+        const dontAskForConfirmation = (isSignedIntoContract && widgetContractPermission && widgetContractPermission[transactions[0].methodName])
+          ? true : false;
```

Comment:
Function calls with non-zero deposits cannot be signed with a limited access key (i.e. "Don't ask again" is not even applicable), and I feel this code does not handle this situation properly.

Reply:
I have added a check also for non-zero deposits now, so that the user is not offered "Don't ask again". Also if a previously approved similar transaction ( same contract, same method name ), now has a deposit, then it will require confirmation.